packages/db/tests/models/test-granule.js

```diff
@@ -0,0 +1,265 @@
+const test = require('ava');
+const cryptoRandomString = require('crypto-random-string');
+
+const {
+  localStackConnectionEnv,
+  getKnexClient,
+  CollectionPgModel,
+  ExecutionPgModel,
+  GranulePgModel,
+} = require('../../dist');
+const {
+  fakeCollectionRecordFactory,
+  fakeExecutionRecordFactory,
+  fakeGranuleRecordFactory,
+} = require('../../dist/test-utils');
+
+const { migrationDir } = require('../../../../lambdas/db-migration');
+
+const testDbName = `granule_${cryptoRandomString({ length: 10 })}`;
+const testDbUser = 'postgres';
+
+test.before(async (t) => {
+  t.context.knexAdmin = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      migrationDir,
+    },
+  });
+  await t.context.knexAdmin.raw(`create database "${testDbName}";`);
+  await t.context.knexAdmin.raw(`grant all privileges on database "${testDbName}" to "${testDbUser}"`);
+
+  t.context.knex = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      PG_DATABASE: testDbName,
+      migrationDir,
+    },
+  });
+
+  // create tables
+  await t.context.knex.migrate.latest();
+
+  t.context.granulePgModel = new GranulePgModel();
+
+  const collectionPgModel = new CollectionPgModel();
+  t.context.collection = fakeCollectionRecordFactory();
+  const collectionResponse = await collectionPgModel.create(
+    t.context.knex,
+    t.context.collection
+  );
+  t.context.collectionCumulusId = collectionResponse[0];
+
+  t.context.executionPgModel = new ExecutionPgModel();
+  const [executionCumulusId] = await t.context.executionPgModel.create(
+    t.context.knex,
+    fakeExecutionRecordFactory()
+  );
+  t.context.executionCumulusId = executionCumulusId;
+});
+
+test.after.always(async (t) => {
+  await t.context.knex.destroy();
+  await t.context.knexAdmin.raw(`drop database if exists "${testDbName}"`);
+  await t.context.knexAdmin.destroy();
+});
+
+test('GranulePgModel.upsert() creates a new running granule', async (t) => {
```

Comment:
Actually looking at the test at  ~L125, this test should be updated to use a running granule.

Reply:
It might also not be a half bad idea not to set a default in the test-util factory (explicit over implicit in the test code) as the test cases aren't *obviously* completed by default.

Reply:
087c65c9ed77c94c9f3802452d139049853d615c